src/components/input-date-picker/input-date-picker.tsx

```diff
@@ -895,35 +861,110 @@ export class InputDatePicker
     }
   };
 
-  /**
-   * Find a date from input string
-   * return false if date is invalid, or out of range
-   *
-   * @param value
-   */
-  private getDateFromInput(value: string): Date | false {
-    if (!this.localeData) {
-      return false;
+  private localizeInputValues(): void {
+    const date = dateFromRange(
+      this.range ? this.startAsDate : this.valueAsDate,
+      this.minAsDate,
+      this.maxAsDate
+    );
+    const endDate = this.range
+      ? dateFromRange(this.endAsDate, this.minAsDate, this.maxAsDate)
+      : null;
+
+    const localizedDate = date && date.toLocaleDateString(this.effectiveLocale);
+    const localizedEndDate = endDate && endDate.toLocaleDateString(this.effectiveLocale);
+
+    localizedDate && this.setInputValue(localizedDate, "start");
+    this.range && localizedEndDate && this.setInputValue(localizedEndDate, "end");
+  }
+
+  private setInputValue = (newValue: string, input: "start" | "end" = "start"): void => {
+    const inputEl = this[`${input}Input`];
+    if (!inputEl) {
+      return;
     }
-    const { separator } = this.localeData;
-    const { day, month, year } = parseDateString(value, this.localeData);
-    const validDay = day > 0;
-    const validMonth = month > -1;
-    const date = new Date(year, month, day);
-    date.setFullYear(year);
-    const validDate = !isNaN(date.getTime());
-    const validLength = value.split(separator).filter((c) => c).length > 2;
-    const validYear = year.toString().length > 0;
+    inputEl.value = newValue;
+  };
+
+  private setRangeValue = (value: Date[] | string): void => {
+    if (!this.range) {
+      return;
+    }
+
+    const { value: oldValue } = this;
+    const oldValueIsArray = Array.isArray(oldValue);
+    const valueIsArray = Array.isArray(value);
+
+    const newStartDate = valueIsArray ? value[0] : "";
+    const newStartDateISO = valueIsArray ? dateToISO(newStartDate) : "";
+    const newEndDate = valueIsArray ? value[1] : "";
+    const newEndDateISO = valueIsArray ? dateToISO(newEndDate) : "";
+
+    // TODO: ensure start and end dates are within min and max
```

Comment:
Are these still pending?

Reply:
These are not, good catch, just removed them.